django/core/checks/caches.py

```diff
@@ -34,43 +44,29 @@ def check_cache_location_not_exposed(app_configs, **kwargs):
                 paths.add(pathlib.Path(staticfiles_dir).resolve())
         else:
             paths = {pathlib.Path(setting).resolve()}
-        for alias in settings.CACHES:
-            cache = caches[alias]
-            if not isinstance(cache, FileBasedCache):
-                continue
-            cache_path = pathlib.Path(cache._dir).resolve()
-            if any(path == cache_path for path in paths):
-                relation = "matches"
-            elif any(path in cache_path.parents for path in paths):
-                relation = "is inside"
-            elif any(cache_path in path.parents for path in paths):
-                relation = "contains"
-            else:
-                continue
-            errors.append(
-                Warning(
-                    f"Your '{alias}' cache configuration might expose your cache "
-                    f"or lead to corruption of your data because its LOCATION "
-                    f"{relation} {name}.",
-                    id="caches.W002",
-                )
-            )
+
+        check_result = FileBasedCache.check(cache, paths, name, alias_name)
+        errors.append(check_result) if check_result else []
```

Comment:
The main idea is to run `check()` on all caches, not to leave the same functions that will run class methods. Check what we did in a similar patches, e.g. 0fb104dda287431f5ab74532e45e8471e22b58c8.